### PR TITLE
Feature: strip ignored characters

### PR DESCRIPTION
**Pull request by rybon**
Adds a method on the resulting query object that when called returns a stripped version of the GraphQL string literal.

### CODE REVIEW

**Comment by rybon** [comment]
@jnwng could you review this one?

File: package.json [hunk 1]
```diff
@@ -25,7 +25,7 @@
     "babel-preset-es2015": "^6.9.0",
     "babel-register": "^6.9.0",
     "chai": "^4.0.2",
-    "graphql": "^14.0.2",
+    "graphql": "^14.3.0",
```
Comment (rybon) [inline package.json]: `stripIgnoredCharacters ` was added in this version and is not available in older versions.

File: src/index.js [hunk 2]
```diff
@@ -1,10 +1,14 @@
 var parser = require('graphql/language/parser');
+var stripIgnoredCharacters = require('graphql').stripIgnoredCharacters;
```
Comment (stalniy) [inline src/index.js]: This is not good! When used on frontend, you will add to the bundle size the whole size of `graphql` dependency.

So, 2 ways to fix this:
1. quick, just `var stripIgnoredCharacters = require('graphql/utilities/stripIgnoredCharacters')`. Similar way as graphql parser is included
2. more complicated, rewrite this module into ES6 and use webpack/rollup to shake out unused dependencies in prebuilt version

Comment (rybon) [reply src/index.js]: Done (1.).

**Comment by jnwng** [comment]
hello @rybon! 

as you've noticed in adding this feature, there's some oddness around the change in a peerDependency should affect the feature set of a module, e.g., adding a new feature that's introduced in a new version of a peerDependency, which attempting to retain backwards-compatibility with other versions. 

in this case, you're attempting to make a runtime check to see if a function exists in a peerDependency. my dilemma here is whether or not to lean into the "breaking change" and bumping the major of `graphql-tag` so that we don't need runtime checks for features, rather, that we can guarantee existence of that feature based on the installation itself.

on principle i want to consider this a little further, and any advice one way or the other would be helpful to me. generally i think there's an unnecessary aversion to bumping major versions (in my head), but it is not unwarranted given the default installation semantics where minor and patch versions are generally updated transparently whereas majors are not.

**Comment by intellix** [comment]
This was closed because we don't want to include the behaviour by default and we should use it manually or? Would be good to get this in ApolloClient somehow

**Pull request closed without merging**